src/ds5/ds5-factory.cpp

```diff
@@ -943,6 +943,31 @@ namespace librealsense
         }
 
         bool compress_while_record() const override { return false; }
+
+        bool contradicts(const stream_profile_interface* a, const std::vector<stream_profile>& others) const override
+        {
+            auto it = std::find_if(others.begin(), others.end(), [](const stream_profile& sp) {
+                return sp.format == RS2_FORMAT_Y16;
+                });
+            bool unrectified_ir_requested = (it != others.end());
+
+            if (auto vid_a = dynamic_cast<const video_stream_profile_interface*>(a))
+            {
+                for (auto request : others)
+                {
+                    if (a->get_framerate() != 0 && request.fps != 0 && (a->get_framerate() != request.fps))
+                        return true;
+                    if (!unrectified_ir_requested)
+                    {
+                        if (vid_a->get_width() != 0 && request.width != 0 && (vid_a->get_width() != request.width))
```

Comment:
PLs check whether Y16/1288 and RGB with VGA/XGA

Reply:
let's talk about this